code/game/objects/items/implants/implant_abductor.dm

```diff
@@ -13,22 +13,51 @@
 		to_chat(imp_in, span_warning("You must wait [timeleft(on_cooldown)*0.1] seconds to use [src] again!"))
 		return
 
-	home.Retrieve(imp_in,1)
+	if(isnull(home) && !link_pad())
+		if(!link_pad())
+			imp_in.balloon_alert(imp_in, "no teleport pads detected!")
+			return
+
+	home.Retrieve(imp_in)
 	on_cooldown = addtimer(VARSET_CALLBACK(src, on_cooldown, null), cooldown , TIMER_STOPPABLE)
 
 /obj/item/implant/abductor/implant(mob/living/target, mob/user, silent = FALSE, force = FALSE)
 	if(..())
-		var/obj/machinery/abductor/console/console
-		if(ishuman(target))
-			var/datum/antagonist/abductor/A = target.mind.has_antag_datum(/datum/antagonist/abductor)
-			if(A)
-				console = get_abductor_console(A.team.team_number)
-				home = console.pad
-
-		if(!home)
-			var/list/consoles = list()
-			for(var/obj/machinery/abductor/console/C in GLOB.machines)
-				consoles += C
-			console = pick(consoles)
+		link_pad()
+		return TRUE
```

Comment:
Github won't let me do a suggestion for silly/stupid reasons so do an early return like

```dm
if(!..())
    return FALSE

link_pad()
return TRUE
```

don't copypaste it won't work because i did fourspaces instead of a tab